src/views/forms/AccountManagementFormsNew/WithdrawForm/WithdrawForm.tsx

```diff
@@ -111,15 +112,14 @@ export const WithdrawForm = () => {
   const freeCollateralBN = useMemo(() => MustBigNumber(freeCollateral?.current), [freeCollateral]);
 
   // Set default values for withdraw from
-  // TODO [onboarding-rewrite]: https://linear.app/dydx/issue/OTE-875/calculate-default-withdrawal-address-for-keplr
-  // if wallet type is cosmos (keplr), change toAddress based on the chainid
-  // B/C cosmos handles multiple chains and each have their own address
   useEffect(() => {
     setTransferType(TransferType.Withdraw);
     setFromChainId(selectedDydxChainId);
     setFromAddress(dydxAddress);
     setFromTokenDenom(usdcDenom);
-    setToAddress(sourceAccount.address);
+    // Cosmos chains connect to the keplr wallet, which has a unique address per chain id
+    const calculatedCosmosAddress = toChainId && cosmosChainAddresses[toChainId];
+    setToAddress(calculatedCosmosAddress ?? sourceAccount.address);
```

Comment:
hmm so instead of setting all of these initial values in a useEffect, these can just be initialized as initial state in useTransfer

Reply:
👍 absolutely agree, i had a comment somewhere about doing this but it probably got lost. 
i have a ticket here: https://linear.app/dydx/issue/OTE-869/optimize-usetransfers to track this work

ideally rn would like to separate PR's by goal though. i feel like once all the functionality is in it will be easier to write the best pattern to optimize this, rather than trying to guess what it should all look like before we have all the features in.

i'll add a TODO comment with this ticket linked to make sure we take care of this if that works for you

Reply:
hmm you could probably do in #1155 directly since that's where you create the WithdrawForm

Reply:
i was hoping to start the deposit form first - cause i think it actually might make the most sense to split the `useTransfers` hook into `useWithdrawals` `useDeposits` so we don't have to flag everything on `TransferType` but i'd want to work through all the gotchas and usecases first to make sure it's the proper refactor